lib/meandro/rules/unused_struct_field.ex

```diff
@@ -0,0 +1,149 @@
+defmodule Meandro.Rule.UnusedStructField do
+  @moduledoc """
+  Finds struct fields that are not used.
+  It has the following assumptions:
+    - As we are not tracking the value of each variable and a variable can be
+      reassigned we don't check the struct name when looking for access to an
+      struct field or modification
+  """
+
+  @behaviour Meandro.Rule
+
+  @impl true
+  def analyze(files_and_asts, _options) do
+    List.flatten(
+      for {file, ast} <- files_and_asts,
+          result <- analyze_module(file, ast, files_and_asts) do
+        result
+      end
+    )
+  end
+
+  @impl true
+  def is_ignored?(_pattern, _ignore_spec) do
+    false
+  end
+
+  defp analyze_module(file, ast, files_and_asts) do
+    struct_info = struct_info(ast)
+
+    case Map.get(struct_info, :fields) do
+      nil ->
+        []
+
+      fields ->
+        module_name = Map.get(struct_info, :module_name)
+        module_aliases = Map.get(struct_info, :module_aliases)
+
+        for field <- fields do
+          unused = is_unused?({field, module_name, module_aliases}, files_and_asts)
+
+          case unused do
+            true ->
+              %Meandro.Rule{
+                file: file,
+                rule: __MODULE__,
+                text: "The field #{field} from the struct #{module_name} is unused"
+              }
+
+            false ->
+              []
+          end
+        end
+    end
+  end
+
+  defp is_unused?({_field, _module, _aliases}, []) do
+    true
+  end
+
+  defp is_unused?({field, module, aliases}, [{_file, ast} | tl]) do
+    functions = Meandro.Util.functions(ast)
+
+    unused_in_functions =
+      for function <- functions do
+        case Macro.prewalk(function, {true, {field, module, aliases}}, &is_unused_in_ast/2) do
+          {_, {true, _}} ->
+            is_unused?({field, module, aliases}, tl)
+
+          {_, {false, _}} ->
+            false
+        end
+      end
+
+    unused = Enum.all?(unused_in_functions)
+
+    case unused do
+      true ->
+        is_unused?({field, module, aliases}, tl)
+
+      false ->
+        false
+    end
+  end
+
+  # looking for fields where the struct is initialized
+  defp is_unused_in_ast(
+         {:%, _, [{:__aliases__, _, aliases}, {:%{}, _, field_list}]} = ast,
+         {result, {field, module, aliases}}
+       ) do
+    result =
+      case List.keyfind(field_list, field, 0, nil) do
```

Comment:
nit: `List.keyfind/4` already defaults to `nil`, there is no need to pass it as an argument here
```suggestion
      case List.keyfind(field_list, field, 0) do
```